From 37bca3fa1833653d8ecbdefb7b2185ecb52e121a Mon Sep 17 00:00:00 2001
Message-Id: <37bca3fa1833653d8ecbdefb7b2185ecb52e121a.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Fri, 26 Sep 2014 15:53:32 +0530
Subject: [PATCH 34/75] REVERTME [VPG]: adf: Disable the buffer offset check
 for now

Relook into this. As of now DRM GEM offset is non-zero for mmap to work.
This check was causing always failing as offset was non zero as expected
by ADF layer

For: GMINL-3547
Change-Id: Ifcfeae16904a091ed347802596bba6ca9ef2d0f4
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/video/adf/adf.c |    5 +++++
 1 file changed, 5 insertions(+)

--- linux.orig/drivers/video/adf/adf.c
+++ linux/drivers/video/adf/adf.c
@@ -1076,7 +1076,10 @@ int adf_format_validate_yuv(struct adf_d
 
 	for (i = 0; i < num_planes; i++) {
 		u32 width = buf->w / (i != 0 ? hsub : 1);
+/* FIXME: */
+#ifndef CONFIG_ADF_INTEL_VLV
 		u32 height = buf->h / (i != 0 ? vsub : 1);
+#endif
 		u8 cpp = adf_format_plane_cpp(buf->format, i);
 		u32 last_line_size;
 
@@ -1099,6 +1102,7 @@ int adf_format_validate_yuv(struct adf_d
 			BUG();
 		}
 
+#ifndef CONFIG_ADF_INTEL_VLV
 		if ((u64) (height - 1) * buf->pitch[i] + last_line_size +
 				buf->offset[i] > buf->dma_bufs[i]->size) {
 			dev_err(&dev->base.dev, "plane %u buffer too small (height = %u, pitch = %u, offset = %u, size = %zu)\n",
@@ -1106,6 +1110,7 @@ int adf_format_validate_yuv(struct adf_d
 					buf->offset[i], buf->dma_bufs[i]->size);
 			return -EINVAL;
 		}
+#endif
 	}
 
 	return 0;
